From foo@baz.org Sun Jun  5 14:33:21 2005
Date: Mon, 20 Jun 2005 21:15:16 -0700
To: Greg KH <gregkh@suse.de>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: devfs: Remove devfs_mk_dir() function from the kernel tree

Removes the devfs_mk_dir() function and all callers of it.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 arch/um/drivers/ubd_kern.c          |    2 --
 drivers/block/acsi_slm.c            |    1 -
 drivers/block/floppy.c              |    2 --
 drivers/block/loop.c                |    2 --
 drivers/block/nbd.c                 |    1 -
 drivers/block/paride/pg.c           |    1 -
 drivers/block/paride/pt.c           |    1 -
 drivers/block/rd.c                  |    2 --
 drivers/block/swim3.c               |    2 --
 drivers/block/sx8.c                 |    2 --
 drivers/block/ub.c                  |    1 -
 drivers/cdrom/sbpcd.c               |    2 --
 drivers/char/ipmi/ipmi_devintf.c    |    2 --
 drivers/char/istallion.c            |    1 -
 drivers/char/lp.c                   |    1 -
 drivers/char/ppdev.c                |    1 -
 drivers/char/pty.c                  |    1 -
 drivers/char/stallion.c             |    1 -
 drivers/char/tipar.c                |    3 ---
 drivers/md/dm-ioctl.c               |    1 -
 drivers/md/md.c                     |    1 -
 drivers/media/dvb/dvb-core/dvbdev.c |    3 ---
 drivers/mmc/mmc_block.c             |    1 -
 drivers/net/wan/cosa.c              |    1 -
 drivers/s390/block/dasd.c           |    3 ---
 drivers/s390/block/xpram.c          |    2 --
 drivers/sbus/char/bpp.c             |    1 -
 drivers/sbus/char/vfc_dev.c         |    1 -
 drivers/video/fbmem.c               |    1 -
 fs/coda/psdev.c                     |    1 -
 include/linux/devfs_fs_kernel.h     |    4 ----
 mm/shmem.c                          |    4 +---
 mm/tiny-shmem.c                     |    3 ---
 33 files changed, 1 insertion(+), 55 deletions(-)

--- gregkh-2.6.orig/arch/um/drivers/ubd_kern.c
+++ gregkh-2.6/arch/um/drivers/ubd_kern.c
@@ -841,7 +841,6 @@ int ubd_init(void)
 {
         int i;
 
-	devfs_mk_dir("ubd");
 	if (register_blkdev(MAJOR_NR, "ubd"))
 		return -1;
 
@@ -855,7 +854,6 @@ int ubd_init(void)
 		char name[sizeof("ubd_nnn\0")];
 
 		snprintf(name, sizeof(name), "ubd_%d", fake_major);
-		devfs_mk_dir(name);
 		if (register_blkdev(fake_major, "ubd"))
 			return -1;
 	}
--- gregkh-2.6.orig/drivers/block/acsi_slm.c
+++ gregkh-2.6/drivers/block/acsi_slm.c
@@ -1005,7 +1005,6 @@ int slm_init( void )
 	BufferP = SLMBuffer;
 	SLMState = IDLE;
 	
-	devfs_mk_dir("slm");
 	for (i = 0; i < MAX_SLM; i++) {
 		devfs_mk_cdev(MKDEV(ACSI_MAJOR, i),
 				S_IFCHR|S_IRUSR|S_IWUSR, "slm/%d", i);
--- gregkh-2.6.orig/drivers/block/floppy.c
+++ gregkh-2.6/drivers/block/floppy.c
@@ -4261,8 +4261,6 @@ static int __init floppy_init(void)
 		motor_off_timer[dr].function = motor_off_callback;
 	}
 
-	devfs_mk_dir("floppy");
-
 	err = register_blkdev(FLOPPY_MAJOR, "fd");
 	if (err)
 		goto out_devfs_remove;
--- gregkh-2.6.orig/drivers/block/loop.c
+++ gregkh-2.6/drivers/block/loop.c
@@ -1277,8 +1277,6 @@ static int __init loop_init(void)
 			goto out_mem3;
 	}
 
-	devfs_mk_dir("loop");
-
 	for (i = 0; i < max_loop; i++) {
 		struct loop_device *lo = &loop_dev[i];
 		struct gendisk *disk = disks[i];
--- gregkh-2.6.orig/drivers/block/nbd.c
+++ gregkh-2.6/drivers/block/nbd.c
@@ -672,7 +672,6 @@ static int __init nbd_init(void)
 	printk(KERN_INFO "nbd: registered device at major %d\n", NBD_MAJOR);
 	dprintk(DBG_INIT, "nbd: debugflags=0x%x\n", debugflags);
 
-	devfs_mk_dir("nbd");
 	for (i = 0; i < nbds_max; i++) {
 		struct gendisk *disk = nbd_dev[i].disk;
 		nbd_dev[i].file = NULL;
--- gregkh-2.6.orig/drivers/block/paride/pg.c
+++ gregkh-2.6/drivers/block/paride/pg.c
@@ -674,7 +674,6 @@ static int __init pg_init(void)
 		err = PTR_ERR(pg_class);
 		goto out_chrdev;
 	}
-	devfs_mk_dir("pg");
 	for (unit = 0; unit < PG_UNITS; unit++) {
 		struct pg *dev = &devices[unit];
 		if (dev->present) {
--- gregkh-2.6.orig/drivers/block/paride/pt.c
+++ gregkh-2.6/drivers/block/paride/pt.c
@@ -971,7 +971,6 @@ static int __init pt_init(void)
 		goto out_chrdev;
 	}
 
-	devfs_mk_dir("pt");
 	for (unit = 0; unit < PT_UNITS; unit++)
 		if (pt[unit].present) {
 			class_device_create(pt_class, NULL, MKDEV(major, unit),
--- gregkh-2.6.orig/drivers/block/rd.c
+++ gregkh-2.6/drivers/block/rd.c
@@ -442,8 +442,6 @@ static int __init rd_init(void)
 		goto out;
 	}
 
-	devfs_mk_dir("rd");
-
 	for (i = 0; i < CONFIG_BLK_DEV_RAM_COUNT; i++) {
 		struct gendisk *disk = rd_disks[i];
 
--- gregkh-2.6.orig/drivers/block/swim3.c
+++ gregkh-2.6/drivers/block/swim3.c
@@ -1019,8 +1019,6 @@ int swim3_init(void)
 	int err = -ENOMEM;
 	int i;
 
-	devfs_mk_dir("floppy");
-
 	swim = find_devices("floppy");
 	while (swim && (floppy_count < MAX_FLOPPIES))
 	{
--- gregkh-2.6.orig/drivers/block/sx8.c
+++ gregkh-2.6/drivers/block/sx8.c
@@ -1672,8 +1672,6 @@ static int carm_init_one (struct pci_dev
 	if (host->flags & FL_DYN_MAJOR)
 		host->major = rc;
 
-	devfs_mk_dir(DRV_NAME);
-
 	rc = carm_init_disks(host);
 	if (rc)
 		goto err_out_blkdev_disks;
--- gregkh-2.6.orig/drivers/block/ub.c
+++ gregkh-2.6/drivers/block/ub.c
@@ -2445,7 +2445,6 @@ static int __init ub_init(void)
 
 	if ((rc = register_blkdev(UB_MAJOR, DRV_NAME)) != 0)
 		goto err_regblkdev;
-	devfs_mk_dir(DEVFS_NAME);
 
 	if ((rc = usb_register(&ub_driver)) != 0)
 		goto err_register;
--- gregkh-2.6.orig/drivers/cdrom/sbpcd.c
+++ gregkh-2.6/drivers/cdrom/sbpcd.c
@@ -5808,8 +5808,6 @@ int __init sbpcd_init(void)
 		return -ENOMEM;
 	}
 
-	devfs_mk_dir("sbp");
-
 	for (j=0;j<NR_SBPCD;j++)
 	{
 		struct cdrom_device_info * sbpcd_infop;
--- gregkh-2.6.orig/drivers/char/ipmi/ipmi_devintf.c
+++ gregkh-2.6/drivers/char/ipmi/ipmi_devintf.c
@@ -872,8 +872,6 @@ static __init int init_ipmi_devintf(void
 		ipmi_major = rv;
 	}
 
-	devfs_mk_dir(DEVICE_NAME);
-
 	rv = ipmi_smi_watcher_register(&smi_watcher);
 	if (rv) {
 		unregister_chrdev(ipmi_major, DEVICE_NAME);
--- gregkh-2.6.orig/drivers/char/istallion.c
+++ gregkh-2.6/drivers/char/istallion.c
@@ -5213,7 +5213,6 @@ int __init stli_init(void)
 		printk(KERN_ERR "STALLION: failed to register serial memory "
 				"device\n");
 
-	devfs_mk_dir("staliomem");
 	istallion_class = class_create(THIS_MODULE, "staliomem");
 	for (i = 0; i < 4; i++) {
 		devfs_mk_cdev(MKDEV(STL_SIOMEMMAJOR, i),
--- gregkh-2.6.orig/drivers/char/lp.c
+++ gregkh-2.6/drivers/char/lp.c
@@ -907,7 +907,6 @@ static int __init lp_init (void)
 		return -EIO;
 	}
 
-	devfs_mk_dir("printers");
 	lp_class = class_create(THIS_MODULE, "printer");
 	if (IS_ERR(lp_class)) {
 		err = PTR_ERR(lp_class);
--- gregkh-2.6.orig/drivers/char/ppdev.c
+++ gregkh-2.6/drivers/char/ppdev.c
@@ -782,7 +782,6 @@ static int __init ppdev_init (void)
 		err = PTR_ERR(ppdev_class);
 		goto out_chrdev;
 	}
-	devfs_mk_dir("parports");
 	for (i = 0; i < PARPORT_MAX; i++) {
 		devfs_mk_cdev(MKDEV(PP_MAJOR, i),
 				S_IFCHR | S_IRUGO | S_IWUGO, "parports/%d", i);
--- gregkh-2.6.orig/drivers/char/pty.c
+++ gregkh-2.6/drivers/char/pty.c
@@ -351,7 +351,6 @@ static int pty_unix98_ioctl(struct tty_s
 
 static void __init unix98_pty_init(void)
 {
-	devfs_mk_dir("pts");
 	ptm_driver = alloc_tty_driver(NR_UNIX98_PTY_MAX);
 	if (!ptm_driver)
 		panic("Couldn't allocate Unix98 ptm driver");
--- gregkh-2.6.orig/drivers/char/stallion.c
+++ gregkh-2.6/drivers/char/stallion.c
@@ -3069,7 +3069,6 @@ static int __init stl_init(void)
  */
 	if (register_chrdev(STL_SIOMEMMAJOR, "staliomem", &stl_fsiomem))
 		printk("STALLION: failed to register serial board device\n");
-	devfs_mk_dir("staliomem");
 
 	stallion_class = class_create(THIS_MODULE, "staliomem");
 	for (i = 0; i < 4; i++) {
--- gregkh-2.6.orig/drivers/char/tipar.c
+++ gregkh-2.6/drivers/char/tipar.c
@@ -507,9 +507,6 @@ tipar_init_module(void)
 		goto out;
 	}
 
-	/* Use devfs with tree: /dev/ticables/par/[0..2] */
-	devfs_mk_dir("ticables/par");
-
 	tipar_class = class_create(THIS_MODULE, "ticables");
 	if (IS_ERR(tipar_class)) {
 		err = PTR_ERR(tipar_class);
--- gregkh-2.6.orig/drivers/md/dm-ioctl.c
+++ gregkh-2.6/drivers/md/dm-ioctl.c
@@ -67,7 +67,6 @@ static int dm_hash_init(void)
 {
 	init_buckets(_name_buckets);
 	init_buckets(_uuid_buckets);
-	devfs_mk_dir(DM_DIR);
 	return 0;
 }
 
--- gregkh-2.6.orig/drivers/md/md.c
+++ gregkh-2.6/drivers/md/md.c
@@ -5086,7 +5086,6 @@ static int __init md_init(void)
 		unregister_blkdev(MAJOR_NR, "md");
 		return -1;
 	}
-	devfs_mk_dir("md");
 	blk_register_region(MKDEV(MAJOR_NR, 0), MAX_MD_DEVS, THIS_MODULE,
 				md_probe, NULL, NULL);
 	blk_register_region(MKDEV(mdp_major, 0), MAX_MD_DEVS<<MdpMinorShift, THIS_MODULE,
--- gregkh-2.6.orig/drivers/media/dvb/dvb-core/dvbdev.c
+++ gregkh-2.6/drivers/media/dvb/dvb-core/dvbdev.c
@@ -302,7 +302,6 @@ int dvb_register_adapter(struct dvb_adap
 
 	printk ("DVB: registering new adapter (%s).\n", name);
 
-	devfs_mk_dir("dvb/adapter%d", num);
 	adap->num = num;
 	adap->name = name;
 	adap->module = module;
@@ -409,8 +408,6 @@ static int __init init_dvbdev(void)
 		goto error;
 	}
 
-	devfs_mk_dir("dvb");
-
 	dvb_class = class_create(THIS_MODULE, "dvb");
 	if (IS_ERR(dvb_class)) {
 		retval = PTR_ERR(dvb_class);
--- gregkh-2.6.orig/drivers/mmc/mmc_block.c
+++ gregkh-2.6/drivers/mmc/mmc_block.c
@@ -555,7 +555,6 @@ static int __init mmc_blk_init(void)
 	if (major == 0)
 		major = res;
 
-	devfs_mk_dir("mmc");
 	return mmc_register_driver(&mmc_driver);
 
  out:
--- gregkh-2.6.orig/drivers/net/wan/cosa.c
+++ gregkh-2.6/drivers/net/wan/cosa.c
@@ -393,7 +393,6 @@ static int __init cosa_init(void)
 		err = -ENODEV;
 		goto out;
 	}
-	devfs_mk_dir("cosa");
 	cosa_class = class_create(THIS_MODULE, "cosa");
 	if (IS_ERR(cosa_class)) {
 		err = PTR_ERR(cosa_class);
--- gregkh-2.6.orig/drivers/s390/block/dasd.c
+++ gregkh-2.6/drivers/s390/block/dasd.c
@@ -2111,9 +2111,6 @@ dasd_init(void)
 
 	dasd_diag_discipline_pointer = NULL;
 
-	rc = devfs_mk_dir("dasd");
-	if (rc)
-		goto failed;
 	rc = dasd_devmap_init();
 	if (rc)
 		goto failed;
--- gregkh-2.6.orig/drivers/s390/block/xpram.c
+++ gregkh-2.6/drivers/s390/block/xpram.c
@@ -439,8 +439,6 @@ static int __init xpram_setup_blkdev(voi
 	if (rc < 0)
 		goto out;
 
-	devfs_mk_dir("slram");
-
 	/*
 	 * Assign the other needed values: make request function, sizes and
 	 * hardsect size. All the minor devices feature the same value.
--- gregkh-2.6.orig/drivers/sbus/char/bpp.c
+++ gregkh-2.6/drivers/sbus/char/bpp.c
@@ -1031,7 +1031,6 @@ static int __init bpp_init(void)
 		instances[idx].opened = 0;
 		probeLptPort(idx);
 	}
-	devfs_mk_dir("bpp");
 	for (idx = 0; idx < BPP_NO; idx++) {
 		devfs_mk_cdev(MKDEV(BPP_MAJOR, idx),
 				S_IFCHR | S_IRUSR | S_IWUSR, "bpp/%d", idx);
--- gregkh-2.6.orig/drivers/sbus/char/vfc_dev.c
+++ gregkh-2.6/drivers/sbus/char/vfc_dev.c
@@ -677,7 +677,6 @@ static int vfc_probe(void)
 		kfree(vfc_dev_lst);
 		return -EIO;
 	}
-	devfs_mk_dir("vfc");
 	instance = 0;
 	for_all_sbusdev(sdev, sbus) {
 		if (strcmp(sdev->prom_name, "vfc") == 0) {
--- gregkh-2.6.orig/drivers/video/fbmem.c
+++ gregkh-2.6/drivers/video/fbmem.c
@@ -1429,7 +1429,6 @@ fbmem_init(void)
 {
 	create_proc_read_entry("fb", 0, NULL, fbmem_read_proc, NULL);
 
-	devfs_mk_dir("fb");
 	if (register_chrdev(FB_MAJOR,"fb",&fb_fops))
 		printk("unable to get major %d for fb devs\n", FB_MAJOR);
 
--- gregkh-2.6.orig/fs/coda/psdev.c
+++ gregkh-2.6/fs/coda/psdev.c
@@ -365,7 +365,6 @@ static int init_coda_psdev(void)
 		err = PTR_ERR(coda_psdev_class);
 		goto out_chrdev;
 	}		
-	devfs_mk_dir ("coda");
 	for (i = 0; i < MAX_CODADEVS; i++) {
 		class_device_create(coda_psdev_class, NULL,
 				MKDEV(CODA_PSDEV_MAJOR,i), NULL, "cfs%d", i);
--- gregkh-2.6.orig/include/linux/devfs_fs_kernel.h
+++ gregkh-2.6/include/linux/devfs_fs_kernel.h
@@ -19,10 +19,6 @@ static inline int devfs_mk_symlink(const
 {
 	return 0;
 }
-static inline int devfs_mk_dir(const char *fmt, ...)
-{
-	return 0;
-}
 static inline void devfs_remove(const char *fmt, ...)
 {
 }
--- gregkh-2.6.orig/mm/shmem.c
+++ gregkh-2.6/mm/shmem.c
@@ -2260,9 +2260,7 @@ static int __init init_tmpfs(void)
 		printk(KERN_ERR "Could not register tmpfs\n");
 		goto out2;
 	}
-#ifdef CONFIG_TMPFS
-	devfs_mk_dir("shm");
-#endif
+
 	shm_mnt = do_kern_mount(tmpfs_fs_type.name, MS_NOUSER,
 				tmpfs_fs_type.name, NULL);
 	if (IS_ERR(shm_mnt)) {
--- gregkh-2.6.orig/mm/tiny-shmem.c
+++ gregkh-2.6/mm/tiny-shmem.c
@@ -33,9 +33,6 @@ static int __init init_tmpfs(void)
 {
 	BUG_ON(register_filesystem(&tmpfs_fs_type) != 0);
 
-#ifdef CONFIG_TMPFS
-	devfs_mk_dir("shm");
-#endif
 	shm_mnt = kern_mount(&tmpfs_fs_type);
 	BUG_ON(IS_ERR(shm_mnt));
 
